src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -2662,22 +2659,12 @@ public void EndConnect(IAsyncResult asyncResult)
             EndPoint remoteEndPoint = null;
             ConnectOverlappedAsyncResult coar;
             MultipleAddressConnectAsyncResult macar;
-            ConnectAsyncResult car;
 
             coar = asyncResult as ConnectOverlappedAsyncResult;
             if (coar == null)
             {
                 macar = asyncResult as MultipleAddressConnectAsyncResult;
-                if (macar == null)
```

Comment:
Does your comment about this being dead code mean that macar is never not-null?  If so, could we just make this a regular cast to MultipleAddressConnectAsyncResult and remove the null check on macar entirely?